docs/protocol/specifications/blockchain.md

```diff
@@ -0,0 +1,613 @@
+
+Witness field       | Type                 | Description
+--------------------|----------------------|----------------
+Destination         | ValueDestination1    | The Destination ("forward pointer") for the value contained in this spend. This can point directly to an Output entry, or to a Mux, which points to Output entries via its own Destinations.
+Arguments           | List<String>         | Arguments for the control program contained in the SpentOutput.
+AnchoredEntry       | Pointer                      | Optional pointer to a single entry of any type, which uniquely identifies that entry as one that can use this one as an `Anchor`.
+
+#### Spend Validation
+
+1. Verify that `SpentOutput` is present in the transaction, but do not validate it.
+2. [Validate program](#program-validation) `SpentOutput.ControlProgram` with the given `Arguments` and the transaction version.
+3. Verify that `SpentOutput.Value` is equal to `Destination.Value`.
+4. [Validate](#value-destination-1-validation) `Destination`.
+5. If the transaction version is 1: verify that the `ExtHash` is the all-zero hash.
+
+### Issuance 1
+
+Field               | Type                 | Description
+--------------------|----------------------|----------------
+Type                | String               | "issuance1"
+Body                | Struct               | See below.
+Witness             | Struct               | See below.
+
+Body field          | Type                            | Description
+--------------------|---------------------------------|----------------
+Anchor              | Pointer<Nonce|Spend1|Issuance1> | Entry that this issuance is anchored to.
+Value               | AssetAmount1           | Asset ID and amount being issued.
+Data                | Hash                   | Hash of the reference data for this entry, or a string of 32 zero-bytes (representing no reference data).
+ExtHash             | [ExtStruct](#extension-struct)| If the transaction version is known, this must be 32 zero-bytes.
+
+Witness field       | Type                                      | Description
+--------------------|-------------------------------------------|----------------
+Destination         | ValueDestination1                         | The Destination ("forward pointer") for the value contained in this spend. This can point directly to an `Output`, or to a `Mux`, which points to `Output` entries via its own `Destinations`.
+AssetDefinition     | [Asset Definition](#asset-definition)     | Asset definition for the asset being issued.
+Arguments           | List<String>                              | Arguments for the control program contained in the SpentOutput.
+AnchoredEntry       | Pointer                                   | Optional pointer to a single entry of any type, which uniquely identifies that entry as one that can use this one as an `Anchor`.
+
+#### Issuance Validation
+
+**Inputs:**
+
+1. Issuance entry,
+2. initial block ID.
```

Comment:
Hm: if the initial block ID is required as input here, I wonder whether it should be specified as input to (at least) tx header validation.

Generally speaking, anything required to validate an entry X is part of the validation state required for _all_ entries that might recursively reach X. Might be worth defining a "validation state" data structure.